crates/miden-objects/src/account/storage/slot/named_slot.rs

```diff
@@ -0,0 +1,79 @@
+use crate::account::storage::slot::SlotNameId;
+use crate::account::{SlotName, StorageSlot};
+
+// TODO(named_slots): Docs + separators for the entire module.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct NamedStorageSlot {
+    /// The name of the storage slot.
+    name: SlotName,
+    /// The cached [`SlotNameId`] of the slot name. These must always be consistent with each
+    /// other.
+    ///
+    /// This is cached so that the `Ord` implementation can use the computed name ID instead of
+    /// having to hash the slot name on every comparison operation.
+    name_id: SlotNameId,
```

Comment:
After reviewing 0xMiden/miden-vm#2287, I tried another approach of caching the ID directly in the slot name. It feels a bit less clean (name and ID are less nicely separated and the API is more unusual), but overall probably result in less hashing because the already computed ID could be taken in some cases.

See https://github.com/0xMiden/miden-base/compare/pgackst-named-storage-slots...pgackst-named-storage-slots-cached-id for a diff.

I think I'm still leaning toward the current approach because it seems a bit cleaner, but happy to go in either direction if anyone has a preference.